### ISSUE TITLE
no pi camera

### ISSUE THREAD

**Issue by leoneo** [kizniche/Mycodo#510]
## Mycodo Issue Report:
  - Specific Mycodo Version: 6.1.4
 
#### Problem Description
Please list: no capture pi camera, camera has been enabled in raspi-config.

 Error 500: Internal Server Error

Something bad happened but it's probably not your fault. Letting the developers know about these issues is crucial to supporting Mycodo. Please submit a new issue on GitHub with the following error traceback (copy the entire traceback):

Error (Full Traceback):

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/picamera/camera.py", line 456, in _init_camera
    self._camera = mo.MMALCamera()
  File "/usr/lib/python3/dist-packages/picamera/mmalobj.py", line 2279, in __init__
    super(MMALCamera, self).__init__()
  File "/usr/lib/python3/dist-packages/picamera/mmalobj.py", line 633, in __init__
    prefix="Failed to create MMAL component %s" % self.component_type)
  File "/usr/lib/python3/dist-packages/picamera/exc.py", line 184, in mmal_check
    raise PiCameraMMALError(status, prefix)
picamera.exc.PiCameraMMALError: Failed to create MMAL component b'vc.ril.camera': Out of memory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/pi/Mycodo/env/lib/python3.5/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/pi/Mycodo/env/lib/python3.5/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/pi/Mycodo/env/lib/python3.5/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/pi/Mycodo/env/lib/python3.5/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/pi/Mycodo/env/lib/python3.5/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/pi/Mycodo/env/lib/python3.5/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/pi/Mycodo/env/lib/python3.5/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/home/pi/Mycodo/mycodo/mycodo_flask/routes_page.py", line 142, in page_camera
    camera_record('photo', mod_camera.unique_id)
  File "/home/pi/Mycodo/mycodo/devices/camera.py", line 85, in camera_record
    with picamera.PiCamera() as camera:
  File "/usr/lib/python3/dist-packages/picamera/camera.py", line 431, in __init__
    self._init_camera(camera_num, stereo_mode, stereo_decimate)
  File "/usr/lib/python3/dist-packages/picamera/camera.py", line 460, in _init_camera
    "Camera is not enabled. Try running 'sudo raspi-config' "
picamera.exc.PiCameraError: Camera is not enabled. Try running 'sudo raspi-config' and ensure that the camera has been enabled.

**Comment by kizniche** [comment]
I'm not sure what the issue is, but my Pi camera works with several of my systems. Can you confirm other software can acquire an image? If not, it's probably a hardware issue. Make sure the camera cable is connected in the correct port and orientation (both on the camera and the Pi board).

**Comment by kizniche** [comment]
Also double-check that in fact the camera has been enabled with ```raspi-config```.

**Comment by leoneo** [comment]
Hi 
Yes it is probably a hardware issue because i can not acquire an image with other software

**Comment by leoneo** [comment]
after change the file  /boot/cmdline.txt for pH data
it appear that the camera works.

**Comment by kizniche** [comment]
Good to hear.

**Comment by PatrickBorkowicz** [comment]
Having a similar issue. Pi Camera is enabled in the raspi-config UI. I verified the camera (rev 1.3) is able to capture stills with:

`libcamera-jpeg -o test.jpg`

When adding the Camera in MyCodo the config has the following alert message:
**Raspberry Pi Camera support doesn't appear to be enabled. Please enable it using the Raspberry Pi configuration or with the command 'sudo raspi-config'**

The video stream does not appear when started.

**System Information**
Mycodo Version: 8.12.9
Python Version: 3.9.2 (default, Mar 12 2021, 04:06:34) [GCC 10.2.1 20210110]
Database Version: 0187ea22dc4b

Running on a Pi 3B with latest OS build as of today (2022-02-01). i.e. ran apt-get update / ugrade.

I am wondering if this is due to a change in the schema of cmdline.txt or similar? It appears raspistill / vid have been deprecated so wondering if I need to add another library?

Also thank you for this amazing gem of an application. I had been searching for something like this for some time. The amount of work invested in is incredible!

**Comment by kizniche** [comment]
>When adding the Camera in MyCodo the config has the following alert message:
> Raspberry Pi Camera support doesn't appear to be enabled. Please enable it using the Raspberry Pi configuration or with the command 'sudo raspi-config'

Did you follow these instructions?

**Comment by PatrickBorkowicz** [comment]
@kizniche I managed to resolve it. Thanks for the quick reply though. 

I had run `sudo raspi-config` and enabled the camera _prior_ to running `sudo apt-get update` and `sudo apt-get upgrade` but not after. I went in to raspi-config again and updated the tool to latest, enabled "Legacy Camera Support", then rebooted. This cleared the warning and I can now access the video stream.

Thanks again.
